### PR TITLE
Migrate to FedEx OAuth (new) api endpoints

### PR DESCRIPTION
**Pull request by javamatte**
3 new expected variables for fedex:
- api_key
- secret_key
- url (prod/sandbox)

### CODE REVIEW

File: .github/workflows/continuousIntegration.yaml [hunk 1]
```diff
@@ -53,10 +53,13 @@ jobs:
         DHL_ECOMMERCE_SOLUTIONS_CLIENT_SECRET: ${{ secrets.DHL_ECOMMERCE_SOLUTIONS_CLIENT_SECRET }}
         DHL_ECOMMERCE_SOLUTIONS_ENVIRONMENT_URL: ${{ secrets.DHL_ECOMMERCE_SOLUTIONS_ENVIRONMENT_URL }}
         FEDEX_ACCOUNT_NUMBER: ${{ secrets.FEDEX_ACCOUNT_NUMBER }}
```
Comment (freshlogic) [inline .github/workflows/continuousIntegration.yaml:55]: I don't see the point in leaving FEDEX_ACCOUNT_NUMBER here. `account_number` is still set in the Bloodhound constructor but doesn't appear to be used. Same goes for `FEDEX_ENVIRONMENT` and `FEDEX_KEY` and `FEDEX_METER_NUMBER` and `FEDEX_PASSWORD`

File: carriers/fedEx.js [hunk 2]
```diff
@@ -80,139 +109,121 @@ function FedEx(options) {
...
+
+                // Add url to carrier tracking page
+                results.url = `https://www.fedex.com/apps/fedextrack/?tracknumbers=${encodeURIComponent(trackingNumber)}`;
 
-            callback(null, results);
+                if (!results.shippedAt && results.deliveredAt) {
+                    results.shippedAt = results.deliveredAt;
+                }
+
+                // console.log(typeof callback);
```
Comment (freshlogic) [inline carriers/fedEx.js:221]: Left some debugging here

File: package.json [hunk 3]
```diff
@@ -40,5 +40,5 @@
         "type": "git",
         "url": "https://github.com/mediocre/bloodhound.git"
     },
-    "version": "1.14.0"
+    "version": "1.15.0"
```
Comment (freshlogic) [inline package.json:43]: We should bump to v2.0.0

File: test/carriers/fedEx.js [hunk 4]
```diff
@@ -35,21 +35,24 @@ describe('FedEx', function() {
 
     // https://www.fedex.com/us/developer/webhelp/ws/2018/US/index.htm#t=wsdvg%2FAppendix_F_Test_Server_Mock_Tracking_Numbers.htm
     describe('fedEx.track', function() {
-        this.retries(3);
+        this.retries(0);
```
Comment (freshlogic) [inline test/carriers/fedEx.js:38]: We can remove this line

File: test/carriers/fedEx.js [hunk 5]
```diff
@@ -35,21 +35,24 @@ describe('FedEx', function() {
...
         this.timeout(20000);
 
         const bloodhound = new Bloodhound({
             fedEx: {
                 account_number: process.env.FEDEX_ACCOUNT_NUMBER,
```
Comment (freshlogic) [inline test/carriers/fedEx.js:43]: Here's another area where old properties should be removed

File: test/index.js [hunk 6]
```diff
@@ -8,10 +8,13 @@ describe('Error handling', function() {
     const bloodhound = new Bloodhound({
         fedEx: {
             account_number: process.env.FEDEX_ACCOUNT_NUMBER,
```
Comment (freshlogic) [inline test/index.js:10]: Another area where old properties can be removed.

File: test/carriers/fedEx.js [hunk 7]
```diff
@@ -35,21 +35,24 @@ describe('FedEx', function() {
 
     // https://www.fedex.com/us/developer/webhelp/ws/2018/US/index.htm#t=wsdvg%2FAppendix_F_Test_Server_Mock_Tracking_Numbers.htm
```
Comment (freshlogic) [inline test/carriers/fedEx.js:36]: Is this still the correct URL to reference?

File: .github/workflows/continuousIntegration.yaml (hunk 1 above)
Comment (javamatte) [reply .github/workflows/continuousIntegration.yaml:55]: 100%. Those are not needed any longer and I'll remove them.

**Pull request merged**